src/core_ocean/shared/mpas_ocn_init_routines.F

```diff
@@ -316,50 +316,65 @@ end subroutine ocn_init_routines_setup_sign_and_index_fields!}}}
 !> \author Doug Jacobsen, Mark Petersen, Todd Ringler
 !> \date   September 2011
 !> \details
-!>  This routine initializes meshScaling, meshScalingDel2, and
-!>   meshScalingDel4
+!>  This routine initializes meshScalingDel2 and meshScalingDel4
 !
 !-----------------------------------------------------------------------
-   subroutine ocn_init_routines_compute_mesh_scaling(meshPool, scaleHmixWithMesh, maxMeshDensity)!{{{
+   subroutine ocn_init_routines_compute_mesh_scaling(meshPool, & !{{{
+                 config_hmix_scaleWithMesh, config_maxMeshDensity, &
+                 config_hmix_use_ref_cell_width, config_hmix_ref_cell_width)
 
       type (mpas_pool_type), intent(inout) :: meshPool
-      logical, intent(in) :: scaleHmixWithMesh
-      real (kind=RKIND), intent(in) :: maxMeshDensity
+      logical, intent(in) :: config_hmix_scaleWithMesh
+      real (kind=RKIND), intent(in) :: config_maxMeshDensity
+      logical, intent(in) :: config_hmix_use_ref_cell_width
+      real (kind=RKIND), intent(in) :: config_hmix_ref_cell_width
 
       integer :: iEdge, cell1, cell2
       integer, pointer :: nEdges
       integer, dimension(:,:), pointer :: cellsOnEdge
+      real (kind=RKIND) :: cellWidth
       real (kind=RKIND), dimension(:), pointer :: meshDensity, meshScalingDel2, meshScalingDel4, meshScaling
+      real (kind=RKIND), dimension(:), pointer :: areaCell
 
       call mpas_pool_get_array(meshPool, 'meshDensity', meshDensity)
       call mpas_pool_get_array(meshPool, 'meshScalingDel2', meshScalingDel2)
       call mpas_pool_get_array(meshPool, 'meshScalingDel4', meshScalingDel4)
-      call mpas_pool_get_array(meshPool, 'meshScaling', meshScaling)
       call mpas_pool_get_array(meshPool, 'cellsOnEdge', cellsOnEdge)
+      call mpas_pool_get_array(meshPool, 'areaCell', areaCell)
 
       call mpas_pool_get_dimension(meshPool, 'nEdges', nEdges)
 
-      !
-      ! Compute the scaling factors to be used in the del2 and del4 dissipation
-      !
-      ! Typical use cases have the minval(meshScaling)==1.
-      ! meshScaling values of approximately 1 indicate the highest resolution of the domain.
-
-      meshScalingDel2(:) = 1.0_RKIND
-      meshScalingDel4(:) = 1.0_RKIND
-      meshScaling(:)     = 1.0_RKIND
-
-      if (scaleHmixWithMesh) then
-         do iEdge = 1, nEdges
-            cell1 = cellsOnEdge(1,iEdge)
-            cell2 = cellsOnEdge(2,iEdge)
-            meshScalingDel2(iEdge) = 1.0_RKIND / ( ((meshDensity(cell1) + meshDensity(cell2) ) / 2.0_RKIND) &
-                                   / maxMeshDensity)**(1.0_RKIND / 4.0_RKIND)  ! goes as dc**1
-            meshScalingDel4(iEdge) = 1.0_RKIND / ( ((meshDensity(cell1) + meshDensity(cell2) ) / 2.0_RKIND) &
-                                   / maxMeshDensity)**(3.0_RKIND / 4.0_RKIND)  ! goes as dc**3
-            meshScaling(iEdge)     = 1.0_RKIND / ( ((meshDensity(cell1) + meshDensity(cell2) ) / 2.0_RKIND) &
```

Comment:
The `meshScaling` variable is identical to `meshScalingDel2`, so I removed `meshScaling`. It was only used in the Leith closure, and I changed it to `meshScalingDel2` there.